aries_cloudagent/multitenant/cache.py

```diff
@@ -0,0 +1,94 @@
+"""Cache for multitenancy profiles."""
+
+import logging
+import sys
+from collections import OrderedDict
+from typing import Optional
+
+from ..core.profile import Profile
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ProfileCache:
+    """Profile cache that caches based on LRU strategy."""
+
+    def __init__(self, capacity: int):
+        """Initialize ProfileCache.
+
+        Args:
+            capacity: The capacity of the cache. If capacity is exceeded
+                      profiles are closed.
+        """
+
+        self.profiles: OrderedDict[str, Profile] = OrderedDict()
+        self.capacity = capacity
+
+    async def _cleanup(self):
+        for (key, profile) in self.profiles.items():
+            # When ref count is 4 we can assume the profile is not referenced
+            # 1 = profiles dict
+            # 2 = self.profiles.items()
+            # 3 = profile above
+            # 4 = sys.getrefcount
+            if sys.getrefcount(profile) <= 4:
+                LOGGER.debug(f"closing profile with id {key}")
+                del self.profiles[key]
+                await profile.close()
+
+                if len(self.profiles) <= self.capacity:
+                    break
```

Comment:
It seems a bit odd that this will always iterate through the profiles, short-circuiting if there is one to be cleaned up and we are now below max capacity. Also `self.profiles.items()` will increase the refcount of the dict itself but not the entries it contains, as I understand it (and according to my test just now).

Reply:
I'm not that happy with this approach myself. I'll see if I can rework it a bit. 

Do you have a suggestion on how to approach this? Do you think the way ledger handles refcounting will also work for here?